### PR TITLE
Feat/dark mode

### LINKED ISSUE #2: [Feature request]: Dark Mode
As @billmn & @owenvoke suggested it think it will be nice to add a dark mode for the website

### PR DESCRIPTION
**Pull request by khatabwedaa**
I added a new logo with white color for the dark mode version you have to toggle between them - resolve #2

### CODE REVIEW

File: resources/sass/_base.scss [hunk 1]
```diff
@@ -84,3 +88,20 @@ p {
 ::selection {
     @apply bg-blue-500 text-white;
 }
+
+::-webkit-scrollbar {
```
Comment (octoper) [inline resources/sass/_base.scss]: I think that the scrollbar needs yo change only on the sidebar so we can do something like that
```css
.scrollbar::-webkit-scrollbar {
    width: 8px;
}

/* Track */
.scrollbar::-webkit-scrollbar-track {
    @apply bg-gray-300 bg-opacity-30;
}

/* Handle */
.scrollbar::-webkit-scrollbar-thumb {
    @apply bg-gray-400 bg-opacity-30;
}

/* Handle on hover */
.scrollbar::-webkit-scrollbar-thumb:hover {
    @apply bg-gray-500;
}
```

Also in ``

File: resources/views/components/docs/sidebar.blade.php [hunk 2]
```diff
@@ -7,7 +7,7 @@
 	x-transition:leave-start="top-0"
 	x-transition:leave-end="-top-full">
 
-	<div class="lg:sticky lg:w-56 xl:w-64 lg:px-0 lg:top-0 overflow-y-auto bg-white lg:h-screen fixed left-0 top-nav z-50 overflow-x-auto text-left px-4 h-screen-24 w-full"
+	<div class="lg:sticky lg:w-56 xl:w-64 lg:px-0 lg:top-0 overflow-y-auto bg-white dark:bg-gray-900 lg:h-screen fixed left-0 top-nav z-50 overflow-x-auto text-left px-4 h-screen-24 w-full"
```
Comment (octoper) [inline resources/views/components/docs/sidebar.blade.php]: Add `scrollbar` utility class to have the customized scrollbar only on sidebar.

Comment (khatabwedaa) [reply resources/views/components/docs/sidebar.blade.php]: Okay, I will Do it

Comment (khatabwedaa) [reply resources/views/components/docs/sidebar.blade.php]: But what's is the issue to make it for the main scrollbar too? 🤔

Comment (octoper) [reply resources/views/components/docs/sidebar.blade.php]: It's not a issue but in most cases it's good to keep the default browser scrollbar on the outer page and just customize the scrollbar in individual components

Comment (khatabwedaa) [reply resources/views/components/docs/sidebar.blade.php]: Okay

Comment (khatabwedaa) [reply resources/views/components/docs/sidebar.blade.php]: Done

File: resources/sass/_base.scss [hunk 3]
```diff
@@ -84,3 +88,20 @@ p {
 ::selection {
     @apply bg-blue-500 text-white;
 }
+
+.scrollbar::-webkit-scrollbar {
+    width: 7px;
+    height: 7px;
```
Comment (octoper) [inline resources/sass/_base.scss]: You can remove the height as it's not needed.

File: resources/sass/_base.scss [hunk 4]
```diff
@@ -84,3 +88,20 @@ p {
...
+}
+
+.scrollbar::-webkit-scrollbar-track {
+    background: #2d3748;
```
Comment (octoper) [inline resources/sass/_base.scss]: Use colors from Tailwind pallet instead of hardcoded hex.

File: resources/views/layouts/base.blade.php [hunk 5]
```diff
@@ -36,27 +36,30 @@
...
-
-        <li>
-            Built with <a href="https://laravel.com" class="text-white font-bold hover:text-white" title="Laravel">Laravel</a>
-            and <a href="https://tailwindcss.com" class="text-white font-bold hover:text-white"
-                   title="Tailwind CSS, a utility-first CSS framework">Tailwind CSS</a>.
-            Logo by <a href="https://twitter.com/caneco" class="text-white font-bold hover:text-white" title="Caneco">Caneco</a>.
-        </li>
-    </ul>
-</footer>
+<body class="dark" {{ $attributes->except(['title', 'description']) }}>
```
Comment (octoper) [inline resources/views/layouts/base.blade.php]: Remove `dark` as it will be added with javascript

Comment (khatabwedaa) [reply resources/views/layouts/base.blade.php]: Done

File: resources/views/_partials/welcome/nav.blade.php [hunk 6]
```diff
@@ -1,8 +1,9 @@
-<header class="flex items-center shadow-sm bg-white h-16 py-4" role="banner">
+<header class="flex items-center shadow-sm bg-white dark:bg-gray-900 py-4" role="banner">
     <div class="container flex items-center mx-auto px-6 space-x-4" >
         <div class="flex items-center">
             <a href="/" title="{{ config('app.name')}} home" class="inline-flex items-center">
-                <img class="h-8" loading="lazy" src="https://laravel-zero.com/assets/img/logo.svg" alt="{{ config('app.name')}} logo" />
+                {{-- <img class="h-8" loading="lazy" src="/assets/img/logo.svg" alt="{{ config('app.name')}} logo" /> --}}
+                <img class="h-8" loading="lazy" src="/assets/img/logo-dark.svg" alt="{{ config('app.name')}} logo" />
```
Comment (octoper) [inline resources/views/_partials/welcome/nav.blade.php]: Remove this with 
```html
<img class="h-8 dark:hidden flex" loading="lazy" src="/assets/img/logo-dark.svg" alt="{{ config('app.name')}} logo" />
                <img class="h-8 hidden dark:flex" loading="lazy" src="/assets/img/logo.svg" alt="{{ config('app.name')}} logo" />
```

Also in `tailwind.config.js` variants 
```diff
- display: ['responsive', 'motion-safe', 'motion-reduce'],
+ display: ['responsive', 'motion-safe', 'motion-reduce', 'dark'],
```

File: resources/views/_partials/nav.blade.php [hunk 7]
```diff
@@ -1,14 +1,15 @@
...
+            <button aria-label="Toggle Documentation Navigation" @click.prevent="toggle()"
+                class="text-gray-500 dark:text-gray-200 focus:text-gray-600 dark:focus:text-gray-400 focus:outline-none">
+                <x-i.menu class="h-6 w-6 fill-current md:h-8 md:w-8"></x-i.menu>
             </button>
         </div>
 
         <div class="flex items-center">
             <a href="/" title="{{ config('app.name')}} home" class="inline-flex items-center">
-                <img class="h-8" loading="lazy" src="/assets/img/logo.svg" alt="{{ config('app.name')}} logo" />
+                <img class="h-8" loading="lazy" src="/assets/img/logo-dark.svg" alt="{{ config('app.name')}} logo" />
```
Comment (octoper) [inline resources/views/_partials/nav.blade.php]: Remove this line with 
```html
<img class="h-8 dark:hidden flex" loading="lazy" src="/assets/img/logo-dark.svg" alt="{{ config('app.name')}} logo" />
                <img class="h-8 hidden dark:flex" loading="lazy" src="/assets/img/logo.svg" alt="{{ config('app.name')}} logo" />
```

**Comment by octoper** [comment]
Also you need to create a `darkmode.js` in `resources/js` folder

```js
window.ToggleDark = () => {
    return {
        mode: "",
        toggle() {
            if (this.mode == "light") {
                this.mode = "dark";
                localStorage.setItem("mode", this.mode);
                document.querySelector("html").classList.add("dark");
            }
            else {
                this.mode = "light";
                localStorage.setItem('mode', this.mode);
                document.querySelector('html').classList.remove('dark');
            }
        },
        created () {
            if (localStorage.getItem('mode') === 'dark') {
                this.mode = 'dark';
                document.querySelector('html').classList.add('dark');
            }
            else {
                this.mode = 'light';
                document.querySelector('html').classList.remove('dark');
            }
        }
    }
}
```
and import it in `app.js` 

```diff
import hljs from 'highlight.js/lib/core';
import php from 'highlight.js/lib/languages/php';
import shell from 'highlight.js/lib/languages/shell';
import bash from 'highlight.js/lib/languages/bash';
import json from 'highlight.js/lib/languages/json';
import diff from 'highlight.js/lib/languages/diff';

import 'alpinejs';
import { listen } from "quicklink";
import docsearch from '@docsearch/js';

import './clipboard';
import './nav';
+ import './darkmode'
```
also in `resources/views/_partials/nav.blade.php` you need to add the below html before 
```html
 <div id="docsearch"></div>
```

```html
<button x-data="ToggleDark()" x-cloak x-init="created()" title="Dark Mode" @click.prevent="toggle()"
        class="ml-6 text-gray-100 dark:text-white text-gray-700" :class="{'text-white': mode == 'dark'}">
    <svg fill="none" stroke="currentColor" class="fill-current h-8 h-8 lg:h-6 lg:h-6"
         stroke-linecap="round" stroke-linejoin="round" :class="{'hidden': mode == 'dark'}" stroke-width="2"
         viewBox="0 0 24 24">
        <path d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"></path>
    </svg>
    <svg fill="none" stroke="currentColor" class="fill-current h-8 h-8 lg:h-6 lg:h-6"
         stroke-linecap="round" stroke-linejoin="round" :class="{'hidden': mode == 'light'}" stroke-width="2"
         viewBox="0 0 24 24">
        <path
            d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z">
        </path>
    </svg>
</button>
```

**Comment by khatabwedaa** [comment]
@octoper Everything works well now, Ready to merge Thanks for the help 😝.

**Comment by octoper** [comment]
Merged it!
Thank you for contributing! @khatabwedaa

**Pull request merged**